src/components/organisms/forms/VendorPriceListForm/VendorPriceListForm.tsx

```diff
@@ -0,0 +1,442 @@
+    }),
+    columnHelper.accessor('word_fee', {
+      header: () => t('vendors.word_fee'),
+      footer: (info) => info.column.id,
+    }),
+    columnHelper.accessor('page_fee', {
+      header: () => t('vendors.page_fee'),
+      footer: (info) => info.column.id,
+    }),
+    columnHelper.accessor('minute_fee', {
+      header: () => t('vendors.minute_fee'),
+      footer: (info) => info.column.id,
+    }),
+    columnHelper.accessor('hour_fee', {
+      header: () => t('vendors.hour_fee'),
+      footer: (info) => info.column.id,
+    }),
+    columnHelper.accessor('minimal_fee', {
+      header: () => t('vendors.minimal_fee'),
+      footer: (info) => info.column.id,
+    }),
+    columnHelper.accessor('id', {
+      header: () => <></>,
+      cell: () => (
+        <div className={classes.iconsContainer}>
+          <Button
+            appearance={AppearanceTypes.Text}
+            icon={Edit}
+            ariaLabel={t('vendors.edit_language_pair')}
+            className={classes.editIcon}
+            // handle modal open
+          />
+          <Button
+            appearance={AppearanceTypes.Text}
+            icon={Delete}
+            ariaLabel={t('vendors.delete')}
+            className={classes.deleteIcon}
+            // handle delete
+          />
+        </div>
+      ),
+    }),
+  ] as ColumnDef<any>[]
+
+  const resetForm = useCallback(() => {
+    reset()
+  }, [reset])
+
+  useEffect(() => {
+    resetForm()
+  }, [resetForm])
+
+  const onSubmit: SubmitHandler<FormValues> = useCallback(
+    async (values) => {
+      const transformedArray = flatMap(
+        values.dst_lang_classifier_value_id,
+        (dstValue) =>
+          map(
+            keys(pickBy(values.skill_id, (value) => value === true)),
+            (key, index) => {
+              return {
+                vendor_id: vendor_id,
+                skill_id: key.replace(/_\d+$/, ''),
+                src_lang_classifier_value_id:
+                  values['src_lang_classifier_value_id'],
+                dst_lang_classifier_value_id: dstValue,
+                character_fee: toNumber(values[`character_fee-${index}`]) || 0,
+                word_fee: toNumber(values[`word_fee-${index}`]) || 0,
+                page_fee: toNumber(values[`page_fee-${index}`]) || 0,
+                minute_fee: toNumber(values[`minute_fee-${index}`]) || 0,
+                hour_fee: toNumber(values[`hour_fee-${index}`]) || 0,
+                minimal_fee: toNumber(values[`minimal_fee-${index}`]) || 0,
+              }
+            }
+          )
+      )
+
+      const payload: UpdatePricesPayload = {
+        data: [...transformedArray],
+      }
+
+      try {
+        await createPrices(payload)
+
+        showNotification({
+          type: NotificationTypes.Success,
+          title: t('notification.announcement'),
+          content: t('success.language_pairs_prices_added'),
+        })
+        resetForm()
+      } catch (errorData) {
```

Comment:
**[blocker}**

If we get validation errors we should be showing them under the corresponding fields inside the modal and we shouldn't close the modal. Currently if the user makes a mistake, they have to start the entire flow again, but they should be allowed to fix the mistakes they made. This also means that if the first error was on step 1, then we should show them the first step, instead of the last.

Considering this, we also have https://github.com/keeleinstituut/tv-tolkevarav/issues/426 task for navigating between the steps, so if you want, you can do the 426 task within this task, but it's not necessary.